cli/src/ui.rs

```diff
@@ -217,6 +285,11 @@ impl Ui {
 
         match self.output {
             UiOutput::Terminal { .. } if io::stdout().is_terminal() => {
+                if self.pager_cmd == BUILTIN_PAGER_NAME.into() {
```

Comment:
nit: `matches!(self.pager_cmd, CommandNameAndArgs::String(BUILTIN_PAGER_NAME))` can also be used. It'll make clearer that `[":builtin"]` isn't a builtin pager specifier.